app/controllers/update_username_controller.rb

```diff
@@ -6,15 +6,13 @@ class UpdateUsernameController < ApplicationController
   before_action :require_signed_in
   respond_to :html, :json
   def index
-    @user = User
   end
 
   def update
-    @username = params['username']
-    user = UserImporter.update_username_for_changed_metawiki_usernames(@username)
-    if @username.blank?
-      flash[:error] = t('update_username.empty_username')
-    elsif user.nil?
+    username = UserImporter.sanitize_username params['username']
+    user = UserImporter.update_username_for_global_id username
+    return if username.blank?
```

Comment:
I think still needs to redirect before returning, or it will error.

Reply:
You mean i should redirect first then i should do other things like checking conditions and flashing messages?

Reply:
I think that would work, since every path through this method should redirect the same way.

Reply:
![image](https://user-images.githubusercontent.com/72390769/221980046-cb633b44-4704-45a3-9347-459cd40328fc.png)


Naah, Doing this is giving internal server error.
the way i am doing is not giving error though

Reply:
Cool i've made required changes